src/main.rs

```diff
@@ -32,28 +32,30 @@ fn generate_short_id() -> String {
         .collect()
 }
 
-/// Get passcode from request's cookie
+// Get passcode from request's cookie
 fn get_req_passcode(req: &Request) -> Result<String> {
     let cookie_val: &str = req
         .get_header(header::COOKIE)
         .ok_or_else(|| anyhow!("No cookie found"))?
         .to_str()?;
 
     // we split at ";" not "; ", in case the cookie is ending with ";"
-    cookie_val
+    let passcode = cookie_val
         .split(';')
-        .find_map(|kv| {
-            let index = kv.find('=')?;
+        .map(|kv| {
+            let index = kv
+                .find('=')
+                .ok_or_else(|| anyhow!("Invalid key-value pair"))?;
```

Comment:
error info inside map will not be populated outside of the function, so it is necessary.